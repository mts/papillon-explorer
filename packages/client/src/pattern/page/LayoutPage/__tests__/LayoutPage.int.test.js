import React from 'react'
import { layoutPageAsync, layoutPageDefault } from './LayoutPage.int.render'

jest.mock('@apollo/react-hooks', () => ({ useQuery: () => {} }))
jest.mock('react-router-dom', () => ({ Link: () => <div>some link</div> }))

const apolloReactHooksMockObject = require('@apollo/react-hooks')

describe('<LayoutPage />', () => {
  const pagesDataObject = {
    data: {
      pages: {
        pages: [{ id: 'layout', name: 'some-name' }],
      },
    },
  }

  afterEach(() => jest.clearAllMocks())

  describe('Snaphot', () => {
    test('must match layoutPageAsync', () => {
      jest.spyOn(apolloReactHooksMockObject, 'useQuery').mockReturnValue(pagesDataObject)
      expect(global.renderToJSON(layoutPageAsync)).toMatchSnapshot()
    })

    test('must match layoutPageDefault', () => {
      expect(global.renderToJSON(layoutPageDefault)).toMatchSnapshot()
    })

    test('must match layoutPageNull', () => {
      jest.spyOn(apolloReactHooksMockObject, 'useQuery').mockReturnValue({})
      expect(global.renderToJSON(layoutPageDefault)).toMatchSnapshot()
    })
  })
})
