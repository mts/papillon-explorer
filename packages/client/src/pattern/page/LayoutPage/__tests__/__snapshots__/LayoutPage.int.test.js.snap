// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutPage /> Snaphot must match layoutPageAsync 1`] = `null`;

exports[`<LayoutPage /> Snaphot must match layoutPageDefault 1`] = `
<div>
  <h2>
    some-name
  </h2>
  <div>
    Under construction
     
    <span
      aria-label="emoji"
      role="img"
    >
      ðŸ™‚
    </span>
    <div>
      some link
    </div>
  </div>
</div>
`;

exports[`<LayoutPage /> Snaphot must match layoutPageNull 1`] = `null`;
